groovy-language-server/src/main/java/com/palantir/ls/groovy/GroovyWorkspaceCompiler.java

```diff
@@ -241,13 +240,22 @@ private void resetCompilationUnit() {
     }
 
     private Set<PublishDiagnosticsParams> parseErrors(ErrorCollector collector) {
-        Map<URI, PublishDiagnosticsParamsBuilder> diagnosticsByFile = Maps.newHashMap();
+        Map<URI, List<Diagnostic>> diagnosticsByFile = Maps.newHashMap();
 
         for (int i = 0; i < collector.getWarningCount(); i++) {
             WarningMessage message = collector.getWarning(i);
-            diagnosticsByFile.computeIfAbsent(workspaceRoot.toUri(),
-                    (value) -> new PublishDiagnosticsParamsBuilder().uri(workspaceRoot.toUri().toString()))
-                    .diagnostic(new DefaultDiagnosticBuilder(message.getMessage(), DiagnosticSeverity.Warning).build());
+            String message1 = message.getMessage();
+            checkNotNull(message1, "message cannot be null");
```

Comment:
if it's null, convert it into empty? because groovy's warningMessage;s message is nullable